### PR TITLE
Add auto-generated `version_last` property to support statements

### PR DESCRIPTION
**Pull request by queengooborg**
This PR adds an auto-generated `version_last` property to support statements, which is added during build time.  This property is something that a Yari middleware generates for MDN docs, but adding it to BCD directly would benefit more consumers and reduce the need for middleware.

### CODE REVIEW

File: scripts/release/build.ts [hunk 1]
```diff
@@ -52,18 +54,63 @@ export const applyMirroring = (feature: WalkOutput): void => {
...
+  )) {
+    if (Array.isArray(supportData)) {
+      for (let i = 0; i < supportData.length; i++) {
+        if (supportData[i].version_removed) {
+          (feature.data as any).__compat.support[browser][i].version_last =
+            getPreviousVersion(
+              browser as BrowserName,
+              supportData[i].version_removed,
+            );
+        }
```
Comment (caugner) [inline scripts/release/build.ts:91]: Can we avoid the duplication between lines 85-91 and 94-98? Also, shouldn't we check for `version_removed` as well below?

File: test/linter/test-versions.ts [hunk 2]
```diff
@@ -234,6 +234,12 @@ const checkVersions = (
           chalk`{bold ${browser}} cannot have a {bold version_added: false} in an array of statements.`,
         );
       }
+
+      if ('version_last' in statement) {
+        logger.error(
+          chalk`{bold version_last} is an automatically generated value and should not be manually defined.`,
```
Comment (caugner) [inline test/linter/test-versions.ts:240]:
```suggestion
          chalk`{bold version_last} is automatically generated and should not be defined manually.`,
```

**Review by caugner — CHANGES_REQUESTED** [review]
LGTM, just one and half nits.

File: scripts/release/build.ts (hunk 1 above)
Comment (queengooborg) [reply scripts/release/build.ts]: They're not exactly duplicated -- one is remapping an array, the other just a single object.  I've made some updates to the code to better reflect this!

File: schemas/compat-data-schema.md [hunk 3]
```diff
@@ -315,26 +315,38 @@ Default values:
...
-  "version_added": "3.5",
+  "version_added": "4",
   "version_removed": true
 }
 ```
 
 Note: many data categories no longer allow for `version_removed` to be set to `true`, as we are working to [improve the quality of the compatibility data](https://github.com/mdn/browser-compat-data/issues/3555).
 
+#### `version_last`
+
```
Comment (queengooborg) [inline schemas/compat-data-schema.md:339]:
```suggestion

> [!NOTE]
> This property is automatically generated at build time.

```

**Comment by queengooborg** [comment]
@caugner had given an LGTM with the changes requested, so I'm going to consider that approval of this PR so that we can get this landed for the next release.  Merging.

**Comment by Elchi3** [comment]
LGTM as well. I am just very surprised to see no unit tests for build.ts at all?. Seems like quite an important thing to have tests for given it decides how the public data export will look like... 
I checked how `version_last` is computed and it looks good to me, but again, surprised to have no unit test for the correct generation of `version_last`.

**Comment by queengooborg** [comment]
> I am just very surprised to see no unit tests for build.ts at all?

I can work on increasing unittest coverage shortly!

**Pull request merged**